pkg/metrics/scraper.go

```diff
@@ -66,10 +70,29 @@ func (s *scraper) ScrapeAndParse() (*NodeCounts, error) {
 
 	var nodeCount int
 	for _, metric := range nodeMetricFamily.GetMetric() {
-		nodeCount += int(metric.GetGauge().GetValue())
+		isMetricForLocal, err := isMetricForLocalCluster(metric)
+		clusterNodeCount := int(metric.GetGauge().GetValue())
+		logrus.Debugf("scraper found nodes: %d, isMetricForLocal: %t, err: %v", clusterNodeCount, isMetricForLocal, err)
+		if err != nil {
+			logrus.Warnf("error when attempting to determine if count was for local cluster: %s, will not include %d nodes in total", err.Error(), clusterNodeCount)
+			continue
+		}
+		if !isMetricForLocal {
+			nodeCount += clusterNodeCount
+		}
+
 	}
 
 	return &NodeCounts{
 		Total: nodeCount,
 	}, nil
 }
+
+func isMetricForLocalCluster(metric *prometheusClient.Metric) (bool, error) {
+	for _, label := range metric.GetLabel() {
+		if label.Name != nil && *label.Name == clusterNameLabel {
+			return label.Value != nil && *label.Value == localClusterID, nil
+		}
+	}
+	return false, fmt.Errorf("unable to determine if metric is for local cluster due to missing label")
```

Comment:
This still needs to get changed. Nodes that are unable to be determined would return false, and then on line 81 would get counted. I would also change this method to be `isMetricForDownstream` and then make line 80 a positive `if`.

Reply:
I don't quite agree. Line 76-79 cause a short circuit in the for loop for cases where we got an error when determining if the metric was for the local cluster. We also have test cases which confirm that this behavior works as expected - see lines 61-94 of scraper_test.go. I'm ok with this approach since IIRC bools have a default value of false. The calling function (in this case, `scrapeAndParse`) is responsible for not trusting the first returned value if an error is present.

I can make a change like you are suggesting, but it's going to be a semantic change. More to the point, I think that the current code accurately meets our requirements - we have a function to determine if nodes are for the local cluster, and only count those nodes if there is 1) no error and 2) the function confirms that it is not for the local cluster.